### PR TITLE
mat2: Init at 0.12.0

### PR DESCRIPTION
**Pull request by nicoonoclaste**
###### Motivation for this change

`mat2` is a useful privacy tool.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
      Not applicable
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
      None (new pkg)
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Things yet to do

- [ ] Check upstream's signature on the release tarball
- [ ] Run `mat2`'s testsuite

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6649,6 +6649,21 @@
...
+  nicoo = {
+    name = "nicoo";
+    email = "nicoo@debian.org";
+
+    github = "nicoo";
+    githubId = 1155801;
+    keys = [{
+      longkeyid = "rsa4096/0xEC9D370872BC7A8C";
+      fingerprint = "E44E 9EA5 4B8E 256A FB73  49D3 EC9D 3708 72BC 7A8C";
+    }, {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:6661]:
```suggestion
    }
    {
```

File: pkgs/tools/misc/mat2/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+python3Packages.buildPythonPackage rec {
+  pname = "mat2";
+  version = "0.12.0";
+
+  # TODO: Verify upstream's signature, to ensure maintainers get an authentic
+  #  archive when importing a new version and setting its hash.
+  # #43233 mentions adding support in Nix's tooling, but nothing seems to exist.
+  srcs = fetchurl {
+    url = "${meta.homepage}/-/archive/${version}/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:11]:
```suggestion
    url = "https://0xacab.org/jvoisin/mat2/-/archive/${version}/${pname}-${version}.tar.gz";
```

File: pkgs/tools/misc/mat2/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    sha256 = "185shmq35y2qj8ydy9l6v53flv536b8hrmv35b6ly22bczfs99yj";
+  };
+
+  propagatedBuildInputs =
+    (with python3Packages; [
+      mutagen
+      pygobject3
+    ]) ++ (with pkgs; [
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:19]:
```suggestion
    ]) ++ [
```

File: pkgs/tools/misc/mat2/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      bubblewrap
+      cairo
+      exiftool
+      ffmpeg
+      librsvg
+      poppler_gi
+      gdk_pixbuf
+    ]);
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:27]:
```suggestion
    ];
```

File: pkgs/tools/misc/mat2/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{ python3Packages, fetchurl, lib, pkgs }:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:1]:
```suggestion
{ python3Packages, fetchurl, lib, ... }:
```

pkgs shouldn't be used as an input and should be replaced with the actual packages.

File: pkgs/tools/misc/mat2/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      cairo
+      exiftool
+      ffmpeg
+      librsvg
+      poppler_gi
+      gdk_pixbuf
+    ]);
+
+  # Test runner seems to have incorrect PATH
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:30]:
```
preCheck = ''
  export PATH=../../..:$PATH
'';
```
can we correct it?

File: pkgs/tools/misc/mat2/default.nix [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  meta = with lib; {
+    homepage = "https://0xacab.org/jvoisin/${pname}";
+    description = "Metadata Anonymisation Toolkit — remove metadata from various media formats";
+
+    license = licenses.lgpl3Plus;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:36]:
```suggestion
    description = "Metadata Anonymisation Toolkit — remove metadata from various media formats";
    license = licenses.lgpl3Plus;
```

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -29913,4 +29913,6 @@ in
   lc3tools = callPackage ../development/tools/lc3tools {};
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  mat2 = callPackage ../tools/misc/mat2 {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:29917]: Please sort this into a fitting category and sort there alphabetically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Aslo please squash all the mat2 commits together.

**Comment by nbraud** [comment]
@SuperSandro2000 This PR is marked “draft”.  Your input is not necessary at this point.

**Comment by SuperSandro2000** [comment]
Draft generally means it is not done yet which could mean something is still broken or feedback is required to finish something. Most people appreciate it if they get some help and early feedback to align a package with the contributing guide before they go down the wrong path for to long. Also PRs undrafted after a while can easily get lost and forgotten by reviewers.

Half of my points are required to get this PR accepted and I just wanted to give you some quick feedback on your work to get it merged quick and fast without doing to many review cycles.

**Comment by nbraud** [comment]
> I just wanted to give you some quick feedback on your work to get it merged fast and without doing to many review cycles.

Again: I appreciate that you think you are being helpful, but forcefully un-resolving “conversations” and forcing a contributor to engage with you, when they clearly communicated that your “feedback” isn't welcome as this stage, is not acceptable.

No means no.

File: pkgs/tools/misc/mat2/default.nix [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+    # External binaries called
+    bubblewrap
+    cairo
+    exiftool
+    ffmpeg
+  ];
+
+  # Ensures mat2 is executed with the right interpreter.
+  patchPhase = "patchShebangs ./mat2";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:34]:
```suggestion
  postPatch = ''
    patchShebangs mat2
  '';
```

File: pkgs/tools/misc/mat2/default.nix [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+  # TODO: Figure out why the testsuite fails with bwrap present
+  checkInputs = [ exiftool ffmpeg ];
+  preCheck = "python3 ./mat2 --check-dependencies";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:38]:
```suggestion

  preCheck = ''
    python3 ./mat2 --check-dependencies
  '';
```

File: pkgs/tools/misc/mat2/default.nix [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+  meta = with lib; {
+    homepage = "https://0xacab.org/jvoisin/${pname}";
+    description = "Metadata Anonymisation Toolkit — remove metadata from various media formats";
+
+    license = licenses.lgpl3Plus;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mat2/default.nix:44]:
```suggestion
    description = "Metadata Anonymisation Toolkit — remove metadata from various media formats";
    license = licenses.lgpl3Plus;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by nbraud** [comment]
@SuperSandro2000 Stop.  Period.

**Pull request closed without merging**